src/deprecated/index.ts

```diff
@@ -0,0 +1,30 @@
+/** This is the place where we keep components that are deprecated.
+ *  We don't recommend using it in production.
+ *  If you already use them, please move to the suggested alternative components
+ *
+ *  But, they are published on npm and you can import them.
+ *  example: import {FormGroup} from '@primer/react/deprecated
+ */
+
+export {default as BorderBox} from '../BorderBox'
+export type {BorderBoxProps} from '../BorderBox'
+export {default as Flex} from '../Flex'
+export type {FlexProps} from '../Flex'
+export {default as Grid} from '../Grid'
+export type {GridProps} from '../Grid'
+export {default as Position, Absolute, Fixed, Relative, Sticky} from '../Position'
+export type {PositionProps, AbsoluteProps, FixedProps, RelativeProps, StickyProps} from '../Position'
+export {default as Dialog} from '../Dialog'
```

Comment:
I don't think we want to deprecate Dialog, as there isn't technically a Dialog2 component yet. I'm not actually sure what's going on with this tbh, but it seems like Dialog2 is just a docs-specific alias for Dialog. So maybe this was deprecated mistakenly? 

cc. @colebemis do you have any historical context on Dialog/Dialog2?

Reply:
Yes! This one gets to live! There is no Dialog1 or Dialog2, there is only Dialog in our codebase 😓

Reply:
Update: @pksjce - we've confirmed that this is a hangover from a previous deprecation. 

We need to do some additional triage into this, so for the time being can we please remove Dialog altogether from this PR?